src/commands/registries/azure/DockerWebhookCreateStep.ts

```diff
@@ -67,19 +68,17 @@ export class DockerWebhookCreateStep extends AzureWizardExecuteStep<IAppServiceW
     }
 
     private async createWebhookForApp(node: RemoteTagTreeItem, site: Site, appUri: string): Promise<Webhook | undefined> {
-        // fields derived from the app service wizard
-        let siteName: string = site.name;
-        let baseName = `webapp${siteName}`;
-        let webhookName: string = baseName;
+        let webhookName: string = site.name;
+        // remove disallowed characters
+        webhookName = webhookName.replace(/[^a-zA-Z0-9]/g, '');
+        // add random chars for uniqueness and to ensure min length is met
+        webhookName += randomUtils.getRandomHexString(6);
+        // trim to max length
+        webhookName = webhookName.substr(0, 50);
+
```

Comment:
Is there a reason we removed the check on whether the name already exists?  I realize it may not be fully actionable by the code, but it feels like we can have info that can either be surfaced (possible collision, etc.). 

The current behavior in case of a namespace clash is too silent, IMO.

Reply:
I think we could add random-ness to the generated webhook name while still comparing it to the known webhook names, until we get a non-match (similar to what was done before).  The difference would be to overlap the randomness with the base name when the base name is too long.  (I.e. don't just add the randomness to the end, only to then have it clipped due to length.)

Reply:
> Is there a reason we removed the check on whether the name already exists? I realize it may not be fully actionable by the code, but it feels like we can have info that can either be surfaced (possible collision, etc.).

The call to list webhooks is not guaranteed to return all webhooks in the first batch. So the logic might not even prevent collisions. More importantly, the existing code added an underscore to the name, which is not allowed for webhook names. Which means the code was not even tested and thus didn't seem worth keeping.

Reply:
> The difference would be to overlap the randomness with the base name when the base name is too long.

Done

> I think we could add random-ness to the generated webhook name while still comparing it to the known webhook names, until we get a non-match (similar to what was done before).

The global uniqueness of the site name and the randomly generated chars should already be more than enough to prevent collisions IMO. With that it mind, it just doesn't seem worth it to make the code more complex because I feel like that also adds risk. For example, any while loop could cause the product to hang if not done properly. Also as mentioned above, the existing loop _did_ cause a bug which was never found.

Reply:
This seems like it should work well enough.